### PR TITLE
DataFrame arithmetic operators __finalize__ call

### PR DESCRIPTION
**Pull request by skwirskj**
- [x] xref #28283 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by phofl** [comment]
You can use xref instead of closes, which shows your pr in the issue history

**Comment by skwirskj** [comment]
@phofl Thank you for letting me know!

**Review by phofl — CHANGES_REQUESTED** [review]
small comments

File: pandas/tests/frame/test_arithmetic.py [hunk 1]
```diff
@@ -1732,6 +1732,18 @@ def test_align_int_fill_bug(self):
         expected = df2 - df2.mean()
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "op", [DataFrame.add, DataFrame.sub, DataFrame.mul, DataFrame.div]
```
Comment (phofl) [inline pandas/tests/frame/test_arithmetic.py:1736]: Cover your changes any additional functions?

File: pandas/tests/frame/test_arithmetic.py [hunk 2]
```diff
@@ -1732,6 +1732,18 @@ def test_align_int_fill_bug(self):
...
+    )
+    def test_ops_retain_attrs(self, op):
+        df = DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
```
[the remark this answers is not in the document]
Comment (skwirskj) [reply pandas/tests/frame/test_arithmetic.py:1739]: done

File: doc/source/whatsnew/v1.4.0.rst [hunk 3]
```diff
@@ -627,6 +627,7 @@ Numeric
 - Bug in :class:`DataFrame` arithmetic ops with a subclass whose :meth:`_constructor` attribute is a callable other than the subclass itself (:issue:`43201`)
 - Bug in arithmetic operations involving :class:`RangeIndex` where the result would have the incorrect ``name`` (:issue:`43962`)
 - Bug in arithmetic operations involving :class:`Series` where the result could have the incorrect ``name`` when the operands having matching NA or matching tuple names (:issue:`44459`)
+- Bug in arithmetic operations involving :class:`DataFrame` where the operation does not propagate metadata such as attributes to the result (:issue:`28283`)
```
[the remark this answers is not in the document]
Comment (skwirskj) [reply doc/source/whatsnew/v1.4.0.rst:630]: done

File: pandas/tests/frame/test_arithmetic.py (hunk 1 above)
Comment (skwirskj) [reply pandas/tests/frame/test_arithmetic.py:1736]: I went through and double checked the functions, these are the four that are impacted.

**Comment by jreback** [comment]
closing as stale if you want continue, pls ping

**Pull request closed without merging**